### ISSUE TITLE
useFetcher in 1.10.0 no longer returns previous data during reload

### ISSUE THREAD

**Issue by dmarkow** [remix-run/remix#5064]
### What version of Remix are you using?

1.10.0

### Steps to Reproduce

1.10.0 brings automatic revalidation to `fetcher.load(...)` calls, which is awesome! I'll be excited to get rid of the many `useEffect` calls I have to reload data.

However, it seems that the data handling logic has changed. In 1.9.0, calling `load` on a fetcher that already had data loaded would go through these states:

```
{ state: "idle", data: myOldData }
// Call fetcher.load("/url") again
{ state: "loading", data: myOldData }
{ state: "idle", data: myNewData}
```

In 1.10.0, it's now:
```
{ state: "idle", data: myOldData }
// Call fetcher.load("/url") again, or submit a form which triggers automatic revalidation, and data is undefined
{ state: "loading", data: undefined }
{ state: "idle", data: myNewData}
```

So while it's loading the new data, the old data is no longer being returned. However, the UI often depends on that old data until the new data is available. At the very least, this ends up flashing empty content until the new data is available, but it can actually break the app.

In one of our examples, we load a checklist on-demand in a modal using `useFetcher`. As soon as we check off an item on the list, which submits a form, the checklist disappears because `fetcher.data` is now undefined. Even worse, because the checklist component unmounts since the checklist data is undefined, the child component that made the form submission also gets unmounted. This seems to cancel the revalidation requests, so now the top level checklist modal component's `useFetcher` gets stuck in `state: "loading"`. 

I can add some `useState` calls to hold the most recent version of the data if this is now the expected behavior, but I don't need to do that with `useLoaderData()` so it feels inconsistent.

### Expected Behavior

`useFetcher` should return the old data until the new data is available, just like `useLoaderData()` already does for the route's data.

### Actual Behavior

`useFetcher` returns `data: undefined` any time it's in a loading state, regardless of previously-loaded data.

**Comment by dmarkow** [comment]
@brophdawg11 Just tried out 1.10.1-pre.0 which seems to include these updates, but still having the issue:

- Calling `fetcher.load("/url")` on a fetcher that already has data still briefly sets data to undefined
- Submitting an action on the page causes any fetchers with loaded data to be undefined

At least for calling fetcher.load again, it's hitting https://github.com/remix-run/remix/blob/dev/packages/remix-react/components.tsx#L1303 which resets `data`. I'll need to dig in more on submitting the action.

If you need a quick demo app of the issue, I can throw one together.

**Comment by brophdawg11** [comment]
Nah sorry `1.10.0.pre.0` was a separate node 14 fix, this will be out any minute in `1.10.0-pre.1`

**Comment by dmarkow** [comment]
Sorry, I need to be more patient! `1.10.1-pre.1` seems to fix the issue. Appreciate the quick turnaround, you guys are awesome.

**Comment by brophdawg11** [comment]
This is released in `1.10.1` 👍
